src/monitor-span-workspace/monitor-span.repository.ts

```diff
@@ -16,50 +16,24 @@ export class MonitorSpanWorkspaceRepository extends Repository<MonitorSpan> {
       .getOne();
   }
 
-  async getSpanByLocIdCompTypeCdBeginOrEndDate(
+  async getSpanByLogicalKey(
     locationId: string,
     componentTypeCode: string,
     spanScaleCode: string,
     beginDate: Date,
     beginHour: number,
-    endDate: Date | null,
-    endHour: number | null,
   ): Promise<MonitorSpan | null> {
-    const query = this.createQueryBuilder('ms')
+     const query = this.createQueryBuilder('ms')
       .where('ms.locationId = :locationId', { locationId })
-      .andWhere('ms.componentTypeCode = :componentTypeCode', { componentTypeCode });
+      .andWhere('ms.componentTypeCode = :componentTypeCode', { componentTypeCode })
+      .andWhere('ms.beginDate = :beginDate', { beginDate })
+      .andWhere('ms.beginHour = :beginHour', { beginHour });
 
     if (spanScaleCode) {
       query.andWhere('ms.spanScaleCode = :spanScaleCode', { spanScaleCode });
     }
```

Comment:
I know this code was already in place, but I feel that if the `spanScaleCode` argument is empty, this should explicitly check for a NULL column value. Otherwise it might match a different row with a non-null span scale code.
```suggestion
      .andWhere('ms.beginHour = :beginHour', { beginHour })
      .andWhere('ms.spanScaleCode = :spanScaleCode', { spanScaleCode: spanScaleCode ?? IsNull() });
```